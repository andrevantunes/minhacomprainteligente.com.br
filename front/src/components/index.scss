/*
 * THIS IS AN AUTOGENERATED FILE
 * DO NOT EDIT IT DIRECTLY
 */
@import "./adapters/Accordion/accordion.styles.scss";
@import "./adapters/ItemButton/item-button.styles.scss";
@import "./basics/AuthenticationRequiredMessage/authentication-required-message.styles.scss";
@import "./basics/CardYoutubeNotification/card-youtube-notification.styles.scss";
@import "./basics/Carousel/carousel.styles.scss";
@import "./basics/CartCloser/cart-closer.styles.scss";
@import "./basics/CartSummary/cart-summary.styles.scss";
@import "./basics/CheckoutProduct/checkout-product.styles.scss";
@import "./basics/ComponentItem/component-item.styles.scss";
@import "./basics/ComponentList/component-list.styles.scss";
@import "./basics/Counters/counters.styles.scss";
@import "./basics/DismissedWrapper/dismissed-wrapper.styles.scss";
@import "./basics/Grid/grid.styles.scss";
@import "./basics/GroupNavigation/group-navigation.styles.scss";
@import "./basics/HostHeader/host-header.styles.scss";
@import "./basics/InfoButton/info-button.styles.scss";
@import "./basics/LoginForm/login-form.styles.scss";
@import "./basics/NavigationItem/navigation-item.styles.scss";
@import "./basics/PixSection/pix-section.styles.scss";
@import "./basics/PopUpButton/pop-up-button.styles.scss";
@import "./basics/Product/product.styles.scss";
@import "./basics/ProductList/product-list.styles.scss";
@import "./basics/ProductManagementList/product-management-list.styles.scss";
@import "./basics/PropertySelection/property-selection.styles.scss";
@import "./basics/ReplacementDashboard/replacement-dashboard.styles.scss";
@import "./basics/ReplacementList/replacement-list.styles.scss";
@import "./basics/RepositionFloatButton/reposition-float-button.styles.scss";
@import "./basics/SaleReportElement/sale-report-element.styles.scss";
@import "./basics/Step/step.styles.scss";
@import "./basics/WithdrawSection/withdraw-section.styles.scss";
@import "./sections/Common/SectionCardFooter/section-card-footer.styles.scss";
@import "./sections/CreditCardSection/credit-card-section.styles.scss";
@import "./sections/SectionBase/section-base.styles.scss";
@import "./sections/SectionTitle/section-title.styles.scss";
@import "./specifics/Auth/auth.styles.scss";
@import "./specifics/EssayCorrectedImage/essay-corrected-image.styles.scss";
@import "./specifics/EssayCorrection/essay-correction.styles.scss";
@import "./specifics/EssayFinalGrade/essay-final-grade.styles.scss";
@import "./specifics/EssayGradeAccordion/essay-grade-accordion.styles.scss";
@import "./specifics/EssayRating/essay-rating.styles.scss";
@import "./specifics/EssaySendForm/essay-send-form.styles.scss";
@import "./specifics/EssayTeacherMarkings/essay-teacher-markings.styles.scss";
@import "./specifics/Header/header.styles.scss";
@import "./specifics/MediumReport/medium-report.styles.scss";
@import "./specifics/ProfileForm/profile-form.styles.scss";
@import "./specifics/ProgressBar/progress-bar.styles.scss";
@import "./specifics/SidebarManager/sidebar-manager.styles.scss";
@import "./specifics/SnackBar/snack-bar.styles.scss";
@import "./specifics/TextPlanMaker/text-plan-maker.styles.scss";
@import "./specifics/TextPlanMakerStep1/text-plan-maker-step1.styles.scss";
@import "./specifics/TextPlanMakerStep7/text-plan-maker-step7.styles.scss";
@import "./specifics/ThemeSwitch/theme-switch.styles.scss";
@import "./templates/AppTemplate/app-template.styles.scss";
@import "./templates/AuthTemplate/auth-template.styles.scss";
@import "./templates/CustomerTemplate/customer-template.styles.scss";
@import "./templates/PageTemplate/page-template.styles.scss";
