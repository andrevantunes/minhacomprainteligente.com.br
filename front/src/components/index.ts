/*
 * THIS IS AN AUTOGENERATED FILE
 * DO NOT EDIT IT DIRECTLY
 */
export * from "./adapters/Accordion";
export * from "./adapters/Breadcrumb";
export * from "./adapters/Button";
export * from "./adapters/Calendar";
export * from "./adapters/ItemButton";
export * from "./adapters/Link";
export * from "./adapters/LinkAdapter";
export * from "./adapters/NavigationBarItem";
export * from "./adapters/Ribo";
export * from "./adapters/Tabs";
export * from "./adapters/Text";
export * from "./adapters/Title";
export * from "./adapters/ToggleButton";
export * from "./basics/Brand";
export * from "./basics/CardYoutubeNotification";
export * from "./basics/Carousel";
export * from "./basics/CartCloser";
export * from "./basics/CartSummary";
export * from "./basics/CheckoutProduct";
export * from "./basics/ComponentItem";
export * from "./basics/ComponentList";
export * from "./basics/Counters";
export * from "./basics/DangerousHTML";
export * from "./basics/DismissedWrapper";
export * from "./basics/Grid";
export * from "./basics/GroupNavigation";
export * from "./basics/HostHeader";
export * from "./basics/InfoButton";
export * from "./basics/LabelList";
export * from "./basics/LoginForm";
export * from "./basics/ModalContent";
export * from "./basics/NavigationItem";
export * from "./basics/NotificationModal";
export * from "./basics/PixSection";
export * from "./basics/PopUpButton";
export * from "./basics/Product";
export * from "./basics/ProductList";
export * from "./basics/ProductManagementList";
export * from "./basics/PropertySelection";
export * from "./basics/RawScript";
export * from "./basics/RepositionFloatButton";
export * from "./basics/SaleReportElement";
export * from "./basics/Step";
export * from "./experimental/Conditional";
export * from "./hocs/WithDateRange";
export * from "./hocs/WithDelay";
export * from "./hocs/WithRouter";
export * from "./sections/Common/HeaderSection";
export * from "./sections/Common/SectionCardFooter";
export * from "./sections/CreditCardSection";
export * from "./sections/SectionBase";
export * from "./sections/SectionTitle";
export * from "./specifics/Auth/AccountUpdateForm";
export * from "./specifics/Auth/AuthHeader";
export * from "./specifics/Auth/AuthModal";
export * from "./specifics/Auth/AuthSocials";
export * from "./specifics/Auth/EnemSubscriptionIdUpdateForm";
export * from "./specifics/Auth/SignInForm";
export * from "./specifics/Auth/SignUpForm";
export * from "./specifics/Auth";
export * from "./specifics/DropdownActionAdapter";
export * from "./specifics/DynamicContent";
export * from "./specifics/EssayCorrectedImage";
export * from "./specifics/EssayCorrection";
export * from "./specifics/EssayFinalGrade";
export * from "./specifics/EssayGradeAccordion";
export * from "./specifics/EssayRating";
export * from "./specifics/EssaySendForm";
export * from "./specifics/EssayTeacherMarkings";
export * from "./specifics/HeadHTML";
export * from "./specifics/Header";
export * from "./specifics/MediumReport";
export * from "./specifics/MyEssaysSection";
export * from "./specifics/ProfileForm";
export * from "./specifics/ProgressBar";
export * from "./specifics/Redirect";
export * from "./specifics/Script";
export * from "./specifics/Seo";
export * from "./specifics/ServiceTermsModal";
export * from "./specifics/SidebarManager";
export * from "./specifics/SnackBar";
export * from "./specifics/TextPlanMaker";
export * from "./specifics/TextPlanMakerStep1";
export * from "./specifics/TextPlanMakerStep2";
export * from "./specifics/TextPlanMakerStep3";
export * from "./specifics/TextPlanMakerStep4";
export * from "./specifics/TextPlanMakerStep5";
export * from "./specifics/TextPlanMakerStep6";
export * from "./specifics/TextPlanMakerStep7";
export * from "./specifics/ThemeSwitch";
export * from "./templates/AppTemplate";
export * from "./templates/AuthTemplate";
export * from "./templates/CustomerTemplate";
export * from "./templates/PageTemplate";
