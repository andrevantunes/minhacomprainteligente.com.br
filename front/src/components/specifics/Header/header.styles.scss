.header {
  $height: 64px;
  height: $height;
  &__container {
    display: flex;
    align-items: center;
    justify-content: center;
    height: $height;
    border-bottom: 2px solid $color-primary-500;
    width: 100%;
    background-color: $color-background-secondary;
    z-index: 1;
    top: 0;
    left: 0;
    position: fixed;
    transition: transform ease 200ms;
    transform: translateY(-100%);
  }

  &__menu-btn {
    position: absolute;
    top: 0;
    left: 0;
    height: $gap-lg-2;
    margin: $spacing-sm;
  }

  &--is-going-up {
    .header__container {
      transform: translateY(0);
    }
  }


  @media screen and (min-width: $breakpoint-lg) {
    display: none;
  }
}

