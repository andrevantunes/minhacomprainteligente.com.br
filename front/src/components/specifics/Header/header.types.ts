import { ComponentType } from "@/types";

interface HeaderProps extends ComponentType {}

export type { HeaderProps };
