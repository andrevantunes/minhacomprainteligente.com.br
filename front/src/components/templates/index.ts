/*
 * THIS IS AN AUTOGENERATED FILE
 * DO NOT EDIT IT DIRECTLY
 */
export * from "./AppTemplate";
export * from "./AuthTemplate";
