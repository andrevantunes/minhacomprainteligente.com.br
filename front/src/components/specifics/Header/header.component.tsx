import type { HeaderProps } from "./header.types";

import classNames from "classnames";
import { ToggleButton } from "@andrevantunes/andrevds";

import { SidebarManager } from "@/components";
import React, { useEffect, useRef, useState } from "react";
import { Brand } from "@/components/basics/Brand";

const HEADER_HEIGHT = 64;

const Header = ({ children, ...props }: HeaderProps) => {
  const handleMenuClick = () => {
    SidebarManager.open();
  };

  const prevScrollY = useRef(0);
  const [goingUp, setGoingUp] = useState(true);

  useEffect(() => {
    const handleScroll = () => {
      const currentScrollY = window.scrollY;
      const isAbovePosition = currentScrollY < HEADER_HEIGHT;
      if (!isAbovePosition && prevScrollY.current < currentScrollY && goingUp) {
        setGoingUp(false);
      }
      if ((isAbovePosition || prevScrollY.current > currentScrollY) && !goingUp) {
        setGoingUp(true);
      }
      prevScrollY.current = currentScrollY;
    };

    window.addEventListener("scroll", handleScroll, { passive: true });

    return () => window.removeEventListener("scroll", handleScroll);
  }, [goingUp]);

  const cn = classNames("header", {
    "header--is-going-up": goingUp,
  });

  return (
    <div {...props} className={cn} data-testid="header">
      <div className="header__container">
        <ToggleButton
          variant="text"
          iconName="menu"
          className="header__menu-btn"
          data-testid="open-sidebar-button"
          onClick={handleMenuClick}
        />
        <Brand />
      </div>
    </div>
  );
};

export default Header;
