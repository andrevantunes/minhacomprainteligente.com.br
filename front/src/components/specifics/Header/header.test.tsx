import { render, fireEvent } from "@testing-library/react";

import Header from "./header.component";
import { SidebarManager } from "@/components";

const makeSut = () => render(<Header />);

describe("<Header>", () => {
  it("should render component", () => {
    const wrapper = makeSut();
    expect(wrapper).toMatchSnapshot();
  });

  it('should has "header header--is-going-up" className on initial render component', () => {
    const wrapper = makeSut();
    const header = wrapper.getByTestId("header");
    const className = "header header--is-going-up";
    expect(header.className).toBe(className);
  });

  it("should call Sidebar.open when toggle button is clicked", () => {
    const openSidebarSpy = jest.spyOn(SidebarManager, "open");
    const wrapper = makeSut();

    const button = wrapper.getByTestId("open-sidebar-button");
    fireEvent.click(button);

    expect(openSidebarSpy).toHaveBeenCalledTimes(1);
  });
});
