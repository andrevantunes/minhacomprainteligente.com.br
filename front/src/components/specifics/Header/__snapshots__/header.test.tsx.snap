// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header header--is-going-up"
        data-testid="header"
      >
        <div
          class="header__container"
        >
          <button
            class="btn btn--neutral btn--size-md toggle-btn header__menu-btn toggle-btn--variant-text"
            data-testid="open-sidebar-button"
            type="button"
          >
            <span
              class="icon btn__icon"
              data-name="menu"
              data-testid="menu"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.05 11H3.95C3.42533 11 3 11.4253 3 11.95V12.05C3 12.5747 3.42533 13 3.95 13H20.05C20.5747 13 21 12.5747 21 12.05V11.95C21 11.4253 20.5747 11 20.05 11Z"
                  fill="currentColor"
                />
                <path
                  d="M20.05 16H3.95C3.42533 16 3 16.4253 3 16.95V17.05C3 17.5747 3.42533 18 3.95 18H20.05C20.5747 18 21 17.5747 21 17.05V16.95C21 16.4253 20.5747 16 20.05 16Z"
                  fill="currentColor"
                />
                <path
                  d="M20.05 6H3.95C3.42533 6 3 6.42533 3 6.95V7.05C3 7.57467 3.42533 8 3.95 8H20.05C20.5747 8 21 7.57467 21 7.05V6.95C21 6.42533 20.5747 6 20.05 6Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header header--is-going-up"
      data-testid="header"
    >
      <div
        class="header__container"
      >
        <button
          class="btn btn--neutral btn--size-md toggle-btn header__menu-btn toggle-btn--variant-text"
          data-testid="open-sidebar-button"
          type="button"
        >
          <span
            class="icon btn__icon"
            data-name="menu"
            data-testid="menu"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.05 11H3.95C3.42533 11 3 11.4253 3 11.95V12.05C3 12.5747 3.42533 13 3.95 13H20.05C20.5747 13 21 12.5747 21 12.05V11.95C21 11.4253 20.5747 11 20.05 11Z"
                fill="currentColor"
              />
              <path
                d="M20.05 16H3.95C3.42533 16 3 16.4253 3 16.95V17.05C3 17.5747 3.42533 18 3.95 18H20.05C20.5747 18 21 17.5747 21 17.05V16.95C21 16.4253 20.5747 16 20.05 16Z"
                fill="currentColor"
              />
              <path
                d="M20.05 6H3.95C3.42533 6 3 6.42533 3 6.95V7.05C3 7.57467 3.42533 8 3.95 8H20.05C20.5747 8 21 7.57467 21 7.05V6.95C21 6.42533 20.5747 6 20.05 6Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
